test/e2e/max_pods.go

```diff
@@ -100,6 +100,12 @@ var _ = Describe("MaxPods", func() {
 		By(fmt.Sprintf("Starting additional %v Pods to fully saturate the cluster and trying to start a new one", podsNeededForSaturation))
 		expectNoError(RunRC(config))
 
+		// Make sure that cluster is saturated
```

Comment:
IIUC, you're currently checking that after you try to schedule the max+1 pod, there are max pods running. Which isn't quite the same as checking that before you try to schedule the max+1 pod, there are max pods running.

Anyway, it's not a big deal.